### PR TITLE
feat: Use Conventional Commits for release management/PRs

### PR DESCRIPTION
**Pull request by mofojed**
- Update web-client-ui repository to use Conventional Commits instead of the labelling system we have been using for release management.
- Remove the label check GitHub action/script; we no longer require labels to be added
- Remove `Create Release` action - the version bump/release creation process will be kicked off locally instead of by a GitHub action
- Add GitHub action to enforce conventional commit PR titles
- Update CONTRIBUTING guidelines to detail rules for creating a PR
- Update README to detail how to update release
- Pin the version of ubuntu in all GitHub actions

### CODE REVIEW

File: README.md
Comment (mattrunyon) [inline README.md:86]: We should make this a github action. Idk if we can then enforce permissions so only certain people on the web team can publish. We could work something out using `github.actor` or `github.user` on the action context to limit things if we wanted to.

Benefit to this being an action is I could see an accidental run of this locally without checking out and pulling latest main and I'm not sure what would happen in that case. Duplicates in changelog? Incorrect versioning?

File: CONTRIBUTING.md
Comment (mattrunyon) [inline CONTRIBUTING.md:53]: Will this break if we allow both? Or you just want to encourage the footer so breaking always has a corresponding message?

**Review by mattrunyon — CHANGES_REQUESTED** [review]

File: README.md
Comment (mofojed) [reply README.md:86]: You won't able to push it if you don't have the latest main checked out.

GitHub action doesn't give you the opportunity to confirm the version numbers are correct before it goes through. This is still an improvement on our process, and I wanted to make sure it worked a couple of times manually to iron out any kinks before we let the robots take over.

File: CONTRIBUTING.md
Comment (mofojed) [reply CONTRIBUTING.md:53]: I want to encourage the footer so there is always a corresponding breaking message. Also, `lerna` doesn't seem to process the exclamation mark format as breaking.

**Review by mattrunyon — CHANGES_REQUESTED** [review]

File: .github/workflows/conventional-pr-check.yml
Comment (mofojed) [reply .github/workflows/conventional-pr-check.yml]: True, updated the name. Yea I can disable the status check briefly just to merge this change if we need to.

File: CONTRIBUTING.md
Comment (dsmmcken) [inline CONTRIBUTING.md:47]: Can you list them here for easier reference?

File: .github/workflows/conventional-pr-check.yml
Comment (mattrunyon) [inline .github/workflows/conventional-pr-check.yml:11]: Looks like github didn't run it which makes sense since it's a new workflow. I think it only runs workflows in main (but uses the workflow file from the PR)

As long as we can still merge this PR without the check passing/failing then we should be good.

We should cut a release (using the old style) just before or after merging this

**Review by mattrunyon — APPROVED** [review]

**Review by mattrunyon — CHANGES_REQUESTED** [review]
Hit approve instead of request when I mentioned Don's comment

File: .github/workflows/conventional-pr-check.yml
Comment (mofojed) [reply .github/workflows/conventional-pr-check.yml]: Yes, what I'll do is cut a release, update the required checks, then merge this. All open PRs will need to match.

**Review by dsmmcken — APPROVED** [review]
approved for readme change

**Review by mattrunyon — APPROVED** [review]

**Pull request merged**
